pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -142,3 +195,145 @@ func (a *GCPActuator) getImageID(cd *hivev1.ClusterDeployment, logger log.FieldL
 		return "", errors.New(msg)
 	}
 }
+
+// obtainLease uses the Hive MachinePoolNameLease resource to obtain a unique, single character
+// for use in the name of the machine pool. We are severely restricted on name lengths on GCP
+// and effectively have one character of flexibility with the naming convention originating in
+// the installer.
+func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDeployment, logger log.FieldLogger) (string, bool, error) {
+	logger.Debugf("obtaining lease for pool: %s", pool.Name)
+
+	leases := &hivev1.MachinePoolNameLeaseList{}
+	err := a.client.List(context.TODO(), leases, client.InNamespace(pool.Namespace),
+		client.MatchingLabels(map[string]string{constants.ClusterDeploymentNameLabel: cd.Name}))
+	if err != nil {
+		return "", false, err
+	}
+	logger.Debugf("found %d leases in namespace", len(leases.Items))
+	for _, l := range leases.Items {
+		logger.Debugf("checking lease: %s", l.Name)
+		for _, ownerRef := range l.OwnerReferences {
+			logger.Debugf("checking owner ref: %v", ownerRef)
+			if ownerRef.APIVersion == "hive.openshift.io/v1" &&
+				ownerRef.Kind == "MachinePool" &&
+				ownerRef.Name == pool.Name {
+
+				logger.Debugf("machine pool already has lease: %s", l.Name)
+
+				// Ensure the lease name is in the format we expect, we know everything up to
+				// the last character.
+				expectedLeasePrefix := fmt.Sprintf("%s-", cd.Spec.ClusterMetadata.InfraID)
+				if l.Name[0:len(l.Name)-1] != expectedLeasePrefix {
+					return "", false, fmt.Errorf("lease %s did not match expected lease name format (%s[CHAR])", l.Name, expectedLeasePrefix)
+				}
+
+				return string(l.Name[len(l.Name)-1]), false, nil
+			}
+		}
+	}
+	logger.Debugf("machine pool does not have a lease yet")
+
+	var leaseRune rune
+	// If the pool.Spec.Name == "worker", we want to preseve this MachinePool's "w" character that
+	// the installer would have selected so we do not cycle all worker nodes.
+	// Despite the separation of pool.Name and pool.Spec.Name, we do know that only one pool will
+	// have pool.Spec.Name worker as we validate that the pool must be named
+	// [clusterdeploymentname]-[pool.spec.name]
+	if pool.Spec.Name == "worker" {
+		leaseRune = 'w'
+		logger.Debug("selecting lease char 'w' for original worker pool")
+	} else {
+		// Pool does not have a lease yet, lookup all currently available lease chars
+		availLeaseChars, err := a.findAvailableLeaseChars(cd, leases)
+		if err != nil {
+			return "", false, err
+		}
+		logger.Debugf("availLeaseChars: %v", availLeaseChars)
```

Comment:
I do not think that this will print anything meaningful. I think it will only print the address of the slice--not the contents.

Reply:
It printed the contents of the slice but not as characters. Removing as that's not useful either. :)